### DISCUSSION TITLE
Can't Install watchman on macOS big sur M1

### DISCUSSION THREAD

**Discussion by EmmanuelObua** [Homebrew/discussions#1284] [Getting started]
# Terminal Results
    brew install watchman
    Updating Homebrew...
    ==> Auto-updated Homebrew!
    Updated 2 taps (shivammathur/php and homebrew/core).
    ==> Updated Formulae
    Updated 1 formula.
    
    Error: watchman dependencies not built for the arm64 CPU architecture:
      python@3.9 was built for x86_64
Any help on this shall be appreciated

**Comment by SMillerDev** [comment]
what do `brew doctor` and `brew config` say?

**Reply by carlocab** [reply]
Also `brew info python@3.9`.

**Reply by EmmanuelObua** [reply]
> brew doctor

    Please note that these warnings are just used to help the Homebrew maintainers
    with debugging if you file an issue. If everything you use Homebrew for is
    working fine: please don't worry or file an issue; just ignore this. Thanks!
    
    Warning: Some installed formulae are deprecated or disabled.
    You should find replacements for the following formulae:
      php@7.2
    
    Warning: "config" scripts exist outside your system or Homebrew directories.
    `./configure` scripts often look for *-config scripts to determine if
    software packages are installed, and which additional flags to use when
    compiling and linking.
    
    Having additional scripts in your path can confuse software installed via
    Homebrew if the config script overrides a system or Homebrew-provided
    script of the same name. We found the following "config" scripts:
      /usr/local/bin/python3.9-config
      /usr/local/bin/python3-config
      /Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9-config
      /Library/Frameworks/Python.framework/Versions/3.9/bin/python3-config
    
    Warning: Unbrewed dylibs were found in /usr/local/lib.
    If you didn't put them there on purpose they could cause problems when
    building Homebrew formulae, and may need to be deleted.
    
    Unexpected dylibs:
      /usr/local/lib/libcrypto.1.1.dylib
      /usr/local/lib/libgflags.2.2.dylib
      /usr/local/lib/libglog.0.dylib
      /usr/local/lib/libsodium.23.dylib
      /usr/local/lib/libssl.1.1.dylib
      /usr/local/lib/libtcl8.6.dylib
      /usr/local/lib/libtk8.6.dylib
    
    Warning: Unbrewed header files were found in /usr/local/include.
    If you didn't put them there on purpose they could cause problems when
    building Homebrew formulae, and may need to be deleted.
    
    Unexpected header files:
      /usr/local/include/fakemysql.h
      /usr/local/include/fakepq.h
      /usr/local/include/fakesql.h
      /usr/local/include/itcl.h
      /usr/local/include/itcl2TclOO.h
      /usr/local/include/itclDecls.h
      /usr/local/include/itclInt.h
      /usr/local/include/itclIntDecls.h
      /usr/local/include/itclMigrate2TclCore.h
      /usr/local/include/itclTclIntStubsFcn.h
      /usr/local/include/mysqlStubs.h
      /usr/local/include/odbcStubs.h
      /usr/local/include/pqStubs.h
      /usr/local/include/tcl.h
      /usr/local/include/tclDecls.h
      /usr/local/include/tclOO.h
      /usr/local/include/tclOODecls.h
      /usr/local/include/tclPlatDecls.h
      /usr/local/include/tclThread.h
      /usr/local/include/tclTomMath.h
      /usr/local/include/tclTomMathDecls.h
      /usr/local/include/tdbc.h
      /usr/local/include/tdbcDecls.h
      /usr/local/include/tdbcInt.h
      /usr/local/include/tk.h
      /usr/local/include/tkDecls.h
      /usr/local/include/tkPlatDecls.h
    
    Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
    If you didn't put them there on purpose they could cause problems when
    building Homebrew formulae, and may need to be deleted.
    
    Unexpected '.pc' files:
      /usr/local/lib/pkgconfig/tcl.pc
      /usr/local/lib/pkgconfig/tk.pc
    
    Warning: Unbrewed static libraries were found in /usr/local/lib.
    If you didn't put them there on purpose they could cause problems when
    building Homebrew formulae, and may need to be deleted.
    
    Unexpected static libraries:
      /usr/local/lib/libtclstub8.6.a
      /usr/local/lib/libtkstub8.6.a
    
    Warning: Your Homebrew's prefix is not /usr/local.
    Some of Homebrew's bottles (binary packages) can only be used with the default
    prefix (/usr/local).
    You will encounter build failures with some formulae.
    Please create pull requests instead of asking for help on Homebrew's GitHub,
    Twitter or any other official channels. You are responsible for resolving
    any issues you experience while you are running this
    unsupported configuration.

> brew config

    HOMEBREW_VERSION: 3.1.1
    ORIGIN: https://github.com/Homebrew/brew
    HEAD: dc3e8b485a166a73d6463320d9deaba6236f747c
    Last commit: 3 days ago
    Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
    Core tap HEAD: a9fe5265ae56d3adbe0e1816168858f8609f3364
    Core tap last commit: 2 days ago
    Core tap branch: master
    HOMEBREW_PREFIX: /opt/homebrew
    HOMEBREW_REPOSITORY: /opt/homebrew
    HOMEBREW_CELLAR: /opt/homebrew/Cellar
    HOMEBREW_CASK_OPTS: []
    HOMEBREW_MAKE_JOBS: 8
    Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
    CPU: octa-core 64-bit westmere
    Clang: 12.0 build 1200
    Git: 2.24.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
    Curl: 7.64.1 => /usr/bin/curl
    macOS: 11.0.1-x86_64
    CLT: 12.4.0.0.1.1610135815
    Xcode: 12.4
    Rosetta 2: true

**Reply by EmmanuelObua** [reply]
> `brew info python@3.9`.

    python@3.9: stable 3.9.4
    Interpreted, interactive, object-oriented programming language
    https://www.python.org/
    /opt/homebrew/Cellar/python@3.9/3.9.1_8 (3,971 files, 67.0MB) *
      Poured from bottle on 2021-02-14 at 13:57:02
    From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/python@3.9.rb
    License: Python-2.0
    ==> Dependencies
    Build: pkg-config ✘
    Required: gdbm ✘, mpdecimal ✘, openssl@1.1 ✘, readline ✔, sqlite ✘, xz ✔
    ==> Caveats
    Python has been installed as
      /opt/homebrew/bin/python3
    
    Unversioned symlinks `python`, `python-config`, `pip` etc. pointing to
    `python3`, `python3-config`, `pip3` etc., respectively, have been installed into
      /opt/homebrew/opt/python@3.9/libexec/bin
    
    You can install Python packages with
      pip3 install <package>
    They will install into the site-package directory
      /opt/homebrew/lib/python3.9/site-packages
    
    tkinter is no longer included with this formula, but it is available separately:
      brew install python-tk@3.9
    
    See: https://docs.brew.sh/Homebrew-and-Python
    ==> Analytics
    install: 842,413 (30 days), 2,307,144 (90 days), 4,728,972 (365 days)
    install-on-request: 280,283 (30 days), 776,267 (90 days), 1,386,699 (365 days)
    build-error: 0 (30 days)

**Reply by SMillerDev** [reply]
Are you running brew in a rosetta based terminal?

**Reply by EmmanuelObua** [reply]
Yes.
